main/pipeline/pipeline.cc

```diff
@@ -983,9 +983,11 @@ ast::ParsedFilesOrCancelled typecheck(unique_ptr<core::GlobalState> &gs, vector<
                                                            file.data(igs).path());
                                     }
                                     // Stream out errors
-                                    threadResult.counters = getAndClearThreadCounters();
-                                    resultq->push(move(threadResult), processedByThread);
-                                    processedByThread = 0;
+                                    if (processedByThread >= 10) {
```

Comment:
why?

Reply:
I was having Patrick measure overhead on pay-server. We were seeing a 2s regression, but I think it was due to other changes on master rather than this change.